tests/backup/backup_test.go

```diff
@@ -1048,6 +1047,361 @@ var _ = Describe("{ShareBackupWithUsersAndGroups}", func() {
 	})
 })
 
+var _ = Describe("{ShareLargeNumberOfBackupsWithLargeNumberOfUsers}", func() {
+	numberOfUsers := getEnvInt(usersToBeCreated, "200")
+	numberOfGroups := getEnvInt(groupsToBeCreated, "100")
+	groupSize := getEnvInt(maxUsersInGroup, "2")
+	numberOfBackups := getEnvInt(maxBackupsToBeCreated, "100")
+	timeBetweenConsecutiveBackups := 4 * time.Second
+	users := make([]string, 0)
+	groups := make([]string, 0)
+	backupNames := make([]string, 0)
+	numberOfSimultaneousBackups := 20
+	var contexts []*scheduler.Context
+	labelSelectors := make(map[string]string)
+	var backupLocationUID string
+	var cloudCredUID string
+	var CloudCredUidList []string
+	var appContexts []*scheduler.Context
+	userContexts := make([]context.Context, 0)
+	var bkpNamespaces []string
+	var clusterUid string
+	var clusterStatus api.ClusterInfo_StatusInfo_Status
+	var customBackupLocationName string
+	var credName string
+	var chosenUser string
+	bkpNamespaces = make([]string, 0)
+	providers := getProviders()
+
+	JustBeforeEach(func() {
+		StartTorpedoTest("ShareLargeNumberOfBackupsWithLargeNumberOfUsers",
+			"Share large number of backups to large number of users", nil, 82941)
+		log.InfoD("Deploy applications")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", taskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = appReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				bkpNamespaces = append(bkpNamespaces, namespace)
+			}
+		}
+	})
+	It("Share all backups at cluster level with a user group and revoke it and validate", func() {
+		Step("Validate applications and get their labels", func() {
+			log.InfoD("Validate applications")
+			ValidateApplications(contexts)
+		})
+
+		Step("Create Users", func() {
+			log.InfoD("Creating %d users to be added to the group", numberOfUsers)
+			var wg sync.WaitGroup
+			for i := 1; i <= numberOfUsers; i++ {
+				userName := fmt.Sprintf("testuser%v", i)
+				firstName := fmt.Sprintf("FirstName%v", i)
+				lastName := fmt.Sprintf("LastName%v", i)
+				email := fmt.Sprintf("testuser%v@cnbu.com", i)
+				wg.Add(1)
+				go func(userName, firstName, lastName, email string) {
+					err := backup.AddUser(userName, firstName, lastName, email, "Password1")
+					log.FailOnError(err, "Failed to create user - %s", userName)
+					users = append(users, userName)
+					wg.Done()
```

Comment:
make defer wg.Done() as first statement in the function

Reply:
Done